packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -42,11 +42,10 @@ export type StylesheetFactory = (
 ) => string;
 
 /**
- * The list of stylesheets associated with a template. Each entry is either a StylesheetFactory or a
- * TemplateStylesheetFactory a given stylesheet depends on other external stylesheets (via
- * the @import CSS declaration).
+ * The list of stylesheets associated with a template. Each entry is either a `StylesheetFactory` or
+ * an array of factories that a given stylesheet depends on via CSS `@import` declarations.
  */
-export type TemplateStylesheetFactories = Array<StylesheetFactory | TemplateStylesheetFactories>;
+export type Stylesheets = Array<StylesheetFactory | Stylesheets>;
```

Comment:
I dislike that `StylesheetFactory` is still user-visible here, although it's not a big deal. Feel free to rename to `Stylesheet`; it looks like it won't conflict anywhere.

At some point in the future we really need to make this thing opaque; users shouldn't even be aware that it's a function that takes arguments.

Reply:
The trouble is that the way stylesheets work is they ask us for a thing (`import stylesheet from './style.css'`), and then give us that same thing back (`static stylesheets = [stylesheet]`). So we could give them `unknown` or something vague as the type, but then we'd also _receive_ `unknown`, which makes our own validation harder. Add to that the fact that they can figure out the interface, if they really want to, by inspecting the import at runtime, and there's not really much gained by hiding the type. If all users are doing is providing the stylesheets to the `stylesheets` prop, then I don't think they'll care what the type is, anyway. They'll just care that the type checking passes.